base/testfiles/github-0528.tlg

```diff
@@ -0,0 +1,15 @@
+This is a generated file for the LaTeX2e validation system.
+Don't change this file in any respect.
+(my{file}$with~sp&cials.tex#
+hello :)
+)
+! Illegal parameter number in definition of \reserved@a.
+<to be read again> 
+                   }
+l. ...\texttt{\input{my{file}$with~sp&cials.tex#}}
+You meant to type ## instead of #, right?
+Or maybe a } was forgotten somewhere earlier, and things
+are all screwed up? I'm going to assume that you meant ##.
+(my{file}$with~sp&cials.tex#
```

Comment:
that's really the desired outcome, guess it probably is if we want to stay lean  and simple.

Reply:
Yes, I just added it as a test because it was mentioned in https://github.com/latex3/latex2e/issues/523#issuecomment-792623214, but to change that we'd need to change how `\text@command` works, so it's more of an assertion that it's working as intended